### PR TITLE
systemd-lock-handler: init at 2.4.1

### PR DESCRIPTION
**Pull request by MatthewCroughan**
###### Description of changes

Adds [systemd-lock-handler](https://git.sr.ht/~whynothugo/systemd-lock-handler) to Nixpkgs

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/services/system/systemd-lock-handler/default.nix [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+  cfg = config.services.systemd-lock-handler;
+in {
+  options.services.systemd-lock-handler = {
+    enable = mkEnableOption (lib.mdDoc "systemd-lock-handler");
+    package = mkOption {
+      default = pkgs.systemd-lock-handler;
+      defaultText = literalExpression "pkgs.systemd-lock-handler";
+      type = types.package;
+      description = lib.mdDoc "systemd-lock-handler package to use.";
+    };
```
Comment (liff) [inline nixos/modules/services/system/systemd-lock-handler/default.nix:17]: There seems to be a helper, like `mkEnableOption`, for this:

```suggestion
    package = mkPackageOption pkgs "systemd-lock-handler" { };
```

File: nixos/modules/services/system/systemd-lock-handler/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    config = mkIf cfg.enable {
```
Comment (liff) [inline nixos/modules/services/system/systemd-lock-handler/default.nix:18]: This is nested inside `options.services.systemd-lock-handler` (hence the build failure), so there should be a `};` before this.

File: nixos/modules/services/system/systemd-lock-handler/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+      systemd.user.services.systemd-lock-handler = {
+        description = "Logind lock event to systemd target translation";
+        wantedBy = [ "default.target" ];
+        serviceConfig = {
+          Documentation = "https://sr.ht/~whynothugo/systemd-lock-handler";
+          Slice = [ "session.slice" ];
+          ExecStart = "${cfg.package}/bin/systemd-lock-handler";
```
Comment (liff) [inline nixos/modules/services/system/systemd-lock-handler/default.nix:25]:
```suggestion
          ExecStart = "${getBin cfg.package}/bin/systemd-lock-handler";
```

File: pkgs/os-specific/linux/systemd-lock-handler/default.nix [hunk 4]
```diff
@@ -0,0 +1,25 @@
+{ lib
+, buildGoModule
+, fetchFromSourcehut
+}:
+
+buildGoModule rec {
+  pname = "systemd-lock-handler";
+  version = "2.4.1";
```
Comment (liff) [inline pkgs/os-specific/linux/systemd-lock-handler/default.nix:8]: Version 2.4.2 has been released.

**Review by liff — CHANGES_REQUESTED** [review]
Should build after fixing the module file.

**Comment by liff** [comment]
Have been looking into using this so would like to see it get in :)

File: pkgs/os-specific/linux/systemd-lock-handler/default.nix (hunk 4 above)
Comment (liff) [reply pkgs/os-specific/linux/systemd-lock-handler/default.nix:8]: Perhaps also add `updateScript`?

```nix
  passthru.updateScript = nix-update-script { };
```

**Comment by gepbird** [comment]
Superseded by https://github.com/NixOS/nixpkgs/pull/259196

**Pull request closed without merging**